### PR TITLE
feat(authentication): better authentication logic

### PR DESCRIPTION
**Pull request by guilbep**
# Describe your changes

For jwt and couchdb users the authentication process had some holes.

- the jwt token was not check against its 'exp' field (a.k.a expiration date)
- the auth token from couchdb was not set

# check every five minutes ( 1000 * 60 * 5 ms) 
 - We check the authentication every five minutes

# Related GitHub issues and pull requests

- Ref: #221

### CODE REVIEW

File: frontend/src/store/UserModule.ts [hunk 1]
```diff
@@ -101,23 +101,28 @@ const actions: ActionTree<UserState, RootState> = {
...
-    context.commit("SET_USER_LOADING");
+    if (!byPassLoading) {
+      context.commit("SET_USER_LOADING");
+    }
     return loginJWT(token)
       .then((response) => {
         context.commit("SET_USER", response.data);
         return response;
       })
+      .catch((response: unknown) => {
```
Comment (blueur) [inline frontend/src/store/UserModule.ts:120]: pas bien !

File: frontend/src/utils/couchdb.ts [hunk 2]
```diff
@@ -63,17 +64,48 @@ export function loginDefault(username: string, password: string): AxiosPromise {
...
+  }
+
+  let response: AxiosResponse;
+  try {
+    response = await getSessionWithBearer(token);
+  } catch (error: AxiosError | unknown) {
+    if (axios.isAxiosError(error)) {
+      switch (error?.response?.status) {
+        case 401:
+          console.log(error);
```
Comment (blueur) [inline frontend/src/utils/couchdb.ts:94]: remove log ?

File: frontend/src/utils/couchdb.ts [hunk 3]
```diff
@@ -63,17 +64,48 @@ export function loginDefault(username: string, password: string): AxiosPromise {
...
   sessionStorage.setItem(SessionStorageKey.Token, token);
-
   // parse the jwt, and update userCTX with custom email claim
   const payload = parseJwt(token);
+  if (!payload.exp) {
+    sessionStorage.removeItem(SessionStorageKey.Token);
+    throw new Error("no exp field in JWT token payload");
+  }
+  const expiredDate = new Date(payload.exp * 1000).getTime();
+  const expiredInHowManySecond = parseInt(
```
Comment (blueur) [inline frontend/src/utils/couchdb.ts:75]: ttlSeconds ?

**Review by blueur — APPROVED** [review]

File: frontend/src/store/UserModule.ts (hunk 1 above)
Comment (guilbep) [reply frontend/src/store/UserModule.ts:120]: Merci typescript

**Pull request merged**